src/AspNet.Security.OpenId/OpenIdAuthenticationHandler.cs

```diff
@@ -276,15 +276,23 @@ protected override async Task HandleChallengeAsync(AuthenticationProperties prop
             Request.Scheme + "://" + Request.Host +
             OriginalPathBase + Options.CallbackPath;
 
+        var identity = "http://specs.openid.net/auth/2.0/identifier_select";
+        object? tmpValue;
+        properties.Parameters.TryGetValue(OpenIdAuthenticationConstants.Parameters.Identity, out tmpValue);
+        if (tmpValue != null)
+        {
+            identity = tmpValue.ToString();
```

Comment:
Is `ToString()` the right thing to do here? Feels like it would be better to check it's actually of type `string` (or a type that definitely gives us a useful value when calling `ToString()`) before using it.